docs/design/paginators.md

```diff
@@ -0,0 +1,297 @@
+# Pagination Design
+
+* **Type**: Design
+* **Author(s)**: Aaron Todd
+
+# Abstract
+
+This document presents a design for how paginated operations will be generated.
+
+Smithy services and operations can be marked with the [paginated trait](https://awslabs.github.io/smithy/1.0/spec/core/behavior-traits.html#paginated-trait) which 
+
+> indicates that an operation intentionally limits the number of results returned in a single response an that multiple invocations might be necessary to retrieve all results
+
+
+Pagination works off of a cursor convention where one of the operational input fields is an optional cursor the service 
+uses to produce the next set of results not yet seen. The operation output has a field that marks the cursor position 
+to be used to get the next set of results in subsequent calls.
+
+# Design
+
+Below is an abbreviated example of the trait in use:
+
+```
+service Lambda {
+    operations: [ListFunctions]
+}
+
+@paginated(
+    inputToken: "Marker",
+    outputToken: "NextMarker",
+    pageSize: "MaxItems",
+    items: "Functions"
+)
+@http(method: "PUT", uri: "/2015-03-31/functions", code: 200)
```

Comment:
nit

uri path probably doesn't need the date snapshot prefix, internal detail.